crypto/s390xcap.c

```diff
@@ -13,16 +13,40 @@
 #include <setjmp.h>
 #include <signal.h>
 #include "internal/cryptlib.h"
-
-extern unsigned long OPENSSL_s390xcap_P[];
+#include "s390x_arch.h"
 
 static sigjmp_buf ill_jmp;
 static void ill_handler(int sig)
 {
     siglongjmp(ill_jmp, sig);
 }
 
-unsigned long OPENSSL_s390x_facilities(void);
+/*-
+ * os-specific function to check if "vector enablement control"-bit and
+ * "AFP register control"-bit in control register 0 are set.
+ */
+static int vx_enabled(void)
+{
+#if defined(OPENSSL_SYS_LINUX)
+    FILE *fd;
+    char buf[4096];
+
+    if ((fd = fopen("/proc/cpuinfo", "r")) == NULL)
```

Comment:
If you examine other "cap" modules, you won't find any attempts to read /proc/cpuinfo. The reason for that is unwillingness to make an assumption that /proc is mounted. I for one would argue in favour of attempting to execute a vector instruction. Similar to way it's done in say ppccap.c...

Reply:
This apporach would also eleminate the above os-specific code. Ill take a look at his.